src/nft/components/details/detailsV2/BuyButton.tsx

```diff
@@ -1,21 +1,84 @@
-import { Trans } from '@lingui/macro'
+import { t, Trans } from '@lingui/macro'
 import { formatNumber } from '@uniswap/conedison/format'
+import { useWeb3React } from '@web3-react/core'
+import { useAccountDrawer } from 'components/AccountDrawer'
 import { ButtonPrimary } from 'components/Button'
+import { ButtonGray } from 'components/Button'
 import Loader from 'components/Icons/LoadingSpinner'
+import Row from 'components/Row'
+import { useNftBalance } from 'graphql/data/nft/NftBalance'
+import { AddToBagIcon, CondensedBagIcon } from 'nft/components/icons'
+import { useBag, useProfilePageState, useSellAsset, useWalletCollections } from 'nft/hooks'
 import { useBuyAssetCallback } from 'nft/hooks/useFetchAssets'
-import { GenieAsset } from 'nft/types'
-import styled from 'styled-components/macro'
+import { useIsAssetInBag } from 'nft/hooks/useIsAssetInBag'
+import { GenieAsset, ProfilePageStateType } from 'nft/types'
+import { MouseEvent, useCallback, useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import styled, { css } from 'styled-components/macro'
+import { BREAKPOINTS } from 'theme'
+import { shallow } from 'zustand/shallow'
 
-import { OfferButton } from './OfferButton'
-import { ButtonStyles } from './shared'
-
-const StyledBuyButton = styled(ButtonPrimary)`
+const ButtonStyles = css`
+  padding: 16px;
   display: flex;
   flex-direction: row;
-  padding: 16px 24px;
   gap: 8px;
   line-height: 24px;
   white-space: nowrap;
+`
+
+const BaseButton = styled(ButtonPrimary)`
+  background: none;
+  border: none;
+  border-radius: 0px;
+
+  ${ButtonStyles}
+`
+
+const ButtonSeparator = styled.div<{ shouldHide: boolean }>`
+  height: 24px;
+  width: 0px;
+  border-left: 0.5px solid #f5f6fc;
+
+  ${({ shouldHide }) => shouldHide && `display: none;`}
+`
+
+const AddToBagButton = styled(BaseButton)<{ isExpanded: boolean }>`
+  width: ${({ isExpanded }) => (isExpanded ? '100%' : 'min-content')};
+  transition: ${({ theme }) => theme.transition.duration.medium};
+  flex-shrink: 0;
+  will-change: width;
+`
+
+const StyledBuyButton = styled(BaseButton)<{ shouldHide: boolean }>`
+  min-width: 0px;
+  transition: ${({ theme }) => theme.transition.duration.medium};
+  will-change: width;
+  overflow: hidden;
+
+  ${({ shouldHide }) =>
+    shouldHide &&
+    css`
+      width: 0px;
+      padding: 0px;
+    `}
+`
+
+const ButtonContainer = styled(Row)<{ dataPage: boolean }>`
+  width: 100%;
+  background-color: ${({ theme }) => theme.accentAction};
+  border-radius: 16px;
+  overflow: hidden;
+  white-space: nowrap;
+
+  @media screen and (min-width: ${BREAKPOINTS.sm}px) {
```

Comment:
I think we should use `xs` here. 100% button looks massive on screens > 396px
![Screen Shot 2023-05-31 at 16 05 53 ](https://github.com/Uniswap/interface/assets/11512321/2121f7c5-7560-4fd3-9ab3-9e8e978207ce)